shop/src/test/java/ro/msg/learning/shop/OrderIntegrationTest.java

```diff
@@ -0,0 +1,144 @@
+package ro.msg.learning.shop;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import ro.msg.learning.shop.domain.Customer;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.OrderService;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static ro.msg.learning.shop.TestingDataSource.*;
+
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK, classes = ShopApplication.class)
+@ActiveProfiles("test")
+public class OrderIntegrationTest {
+
+    @Autowired
+    private MockMvc mvc;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private LocationRepository locationRepository;
+    @Autowired
+    private StockRepository stockRepository;
+    @Autowired
+    private OrderService orderService;
+
+
+    @Before
+    public void setUpTest() throws Exception {
```

Comment:
this will populate the db and "after" will clear it per class. Meaning second test will be affected by what happns in the first test. use beforeEach if test data should have integrity between tests. if not, then dont